### PR TITLE
[libnode] Update to v16.14.0 (Latest LTS).

### PR DESCRIPTION
**Pull request by sunoru**

### CODE REVIEW

File: L/libnode/build_tarballs.jl [hunk 1]
```diff
@@ -3,11 +3,11 @@
 using BinaryBuilder, Pkg
 
 name = "libnode"
-version = v"14.17.3"
+version = v"16.13.1"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://nodejs.org/dist/v14.17.3/node-v14.17.3.tar.gz", "dcbd156506ee79ee48439257626ca0a6db3d7eab8cb0208db6979125ae7d3a95"),
+    ArchiveSource("https://nodejs.org/dist/v16.13.1/node-v16.13.1.tar.gz", "34b23965457fb08a8c62f81e8faf74ea60587cda6fa898e5d030211f5f374cb6"),
```
Comment (giordano) [inline L/libnode/build_tarballs.jl:10]:
```suggestion
    ArchiveSource("https://nodejs.org/dist/v$(version)/node-v$(version).tar.gz", "34b23965457fb08a8c62f81e8faf74ea60587cda6fa898e5d030211f5f374cb6"),
```

Comment (sunoru) [reply L/libnode/build_tarballs.jl:10]: Thanks for the suggestion.

**Comment by sunoru** [comment]
Ah. It looks there is something I need to fix before this can be merged. Sorry

**Comment by sunoru** [comment]
Hi @giordano  can you help me? Thanks.

When compiling locally with BinaryBuilder.jl, I got
```
../src/node_main.cc:92:10: fatal error: sys/auxv.h: No such file or directory
    92 | #include <sys/auxv.h>
       |          ^~~~~~~~~~~~
```

This header file (and its content `getauxval`) is introduced since glibc v2.16, so I checked the glibc version of the building environment. It is new enough (v2.29) but this file does not exist. What should I do?

**Comment by giordano** [comment]
Holidays mean that I missed your question before, sorry!  You can try using `linux/auxvec.h`, see for example https://github.com/JuliaPackaging/Yggdrasil/blob/8be19820afa24ff817185f3c53c4d20786a8add3/F/FLAC/build_tarballs.jl#L19-L21 and https://github.com/JuliaPackaging/Yggdrasil/blob/8be19820afa24ff817185f3c53c4d20786a8add3/F/FLAC/bundled/patches/flac_linux_headers.patch

**Comment by sunoru** [comment]
I was wrong :( `getauxval` still needs to be defined on several platforms, and `linux/auxvec.h` does not provide the function.

Do you know why it is not provided by glibc?

**Comment by giordano** [comment]
We use a very old glibc, see https://github.com/JuliaPackaging/Yggdrasil/blob/09f5b1c1773b5c2da1466bd95ce8ca9eecb9f2a7/RootFS.md#glibc-versions.  @staticfloat is actually working on letting us swap glibc versions more easily, but I don't know what's the ETA for that feature to land.

However I'm surprised it works on x86_64-linux-musl.  Do you have a clue of what would be the difference?

**Comment by sunoru** [comment]
It's just because `sys/auxv.h` is included in the musl C library.

**Comment by sunoru** [comment]
@giordano  I'm wondering if it is OK to directly use existing builds from other sources like these: https://rpmfind.net/linux/rpm2html/search.php?query=nodejs-libs

It seems simpler and many platforms are supported. But there are still some dependency issues to fix and it only works for Linux.

**Comment by giordano** [comment]
Of course `x86_64-linux-musl` doesn't use glibc :facepalm: :facepalm: :facepalm:

File: L/libnode/build_tarballs.jl [hunk 2]
```diff
@@ -3,11 +3,11 @@
...
-version = v"14.17.3"
+version = v"16.14.0"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://nodejs.org/dist/v14.17.3/node-v14.17.3.tar.gz", "dcbd156506ee79ee48439257626ca0a6db3d7eab8cb0208db6979125ae7d3a95"),
+    ArchiveSource("https://nodejs.org/dist/v$(version)/node-v$(version).tar.gz", "29dfce13650f063ff009d18349636333fa4305468b6a8965d442c2e88b1dd60f"),
 ]
 
 host_platform = HostPlatform()
```
Comment (giordano) [inline L/libnode/build_tarballs.jl:13]: I just realised this isn't used at all

**Comment by giordano** [comment]
> I'm wondering if it is OK to directly use existing builds from other sources like these: [rpmfind.net/linux/rpm2html/search.php?query=nodejs-libs](https://rpmfind.net/linux/rpm2html/search.php?query=nodejs-libs)

We occasionally do that when there aren't other options.  This isn't great, but a decent stopgap solution.

> It seems simpler and many platforms are supported. But there are still some dependency issues to fix and it only works for Linux.

Aren't you building only for Linux anyways?

File: L/libnode/build_tarballs.jl (hunk 2 above)
Comment (sunoru) [reply L/libnode/build_tarballs.jl:13]: Ah you are right!

**Comment by sunoru** [comment]
Yes I'm only building for Linux for now so that doesn't matter much. The source code of NodeJS is kind of complicated and difficult to build as a shared library, especially for cross-compilation. BTW, if it is allowed to directly use prebuilt libraries, maybe I can also find those for MacOS and Windows.

**Comment by sunoru** [comment]
@giordano I ended up deciding to use patches. It works now and hopefully can be merged?

**Comment by giordano** [comment]
That's awesome, thanks!

**Comment by sunoru** [comment]
Thank you so much

**Pull request merged**